assets/js/modules/adsense/components/dashboard/DashboardTopEarningPagesWidget.js

```diff
@@ -0,0 +1,140 @@
+
+/**
+ * WordPress dependencies
+ */
+import { __, _x } from '@wordpress/i18n';
+import { compose } from '@wordpress/compose';
+
+/**
+ * Internal dependencies
+ */
+import Data from 'googlesitekit-data';
+import { STORE_NAME as ANALYTICS_STORE } from '../../../analytics/datastore/constants';
+import { STORE_NAME as CORE_USER } from '../../../../googlesitekit/datastore/user/constants';
+import whenActive from '../../../../util/when-active';
+import PreviewTable from '../../../../components/preview-table';
+import { getDataTableFromData, TableOverflowContainer } from '../../../../components/data-table';
+import Layout from '../../../../components/layout/layout';
+import AdSenseLinkCTA from '../../../analytics/components/common/AdSenseLinkCTA';
+import getDataErrorComponent from '../../../../components/notifications/data-error';
+import getNoDataComponent from '../../../../components/notifications/nodata';
+
+const { useSelect } = Data;
+
+function DashboardTopEarningPagesWidget() {
+	const {
+		data,
+		error,
+		loading,
+	} = useSelect( ( select ) => {
+		const store = select( ANALYTICS_STORE );
+		const args = {
+			dateRange: select( CORE_USER ).getDateRange(),
+			dimensions: [ 'ga:pageTitle', 'ga:pagePath' ],
+			metrics: [
+				{ expression: 'ga:adsenseRevenue', alias: 'Earnings' },
+				{ expression: 'ga:adsenseECPM', alias: 'Page RPM' },
+				{ expression: 'ga:adsensePageImpressions', alias: 'Impressions' },
+			],
+			orderby: {
+				fieldName: 'ga:adsenseRevenue',
+				sortOrder: 'DESCENDING',
+			},
+			limit: 10,
+		};
+
+		return {
+			data: store.getReport( args ),
+			error: store.getErrorForSelector( 'getReport', [ args ] ),
+			loading: store.isResolving( 'getReport', [ args ] ),
+		};
+	} );
+
+	if ( loading ) {
+		return (
+			<PreviewTable rows={ 5 } padding />
+		);
+	}
+
+	if ( error ) {
+		// Specifically looking for string "badRequest"
+		if ( 'badRequest' === error?.data?.reason ) {
+			return (
+				<Layout className="googlesitekit-top-earnings-pages" fill>
+					<AdSenseLinkCTA />
+				</Layout>
+			);
+		}
+		return getDataErrorComponent( 'adsense', error.message, false, false, false, error );
+	}
+
+	if ( ! data || ! data[ 0 ]?.data?.rows ) {
+		return getNoDataComponent( __( 'Analytics and AdSense', 'google-site-kit' ) );
+	}
+
+	const headers = [
+		{
+			title: __( 'Top Earning Pages', 'google-site-kit' ),
+			tooltip: __( 'Top Earning Pages', 'google-site-kit' ),
+			primary: true,
+		},
+		{
+			title: __( 'Revenue', 'google-site-kit' ),
+			tooltip: __( 'Revenue', 'google-site-kit' ),
+		},
+	];
+
+	const links = [];
+	const dataMapped = data[ 0 ].data.rows.map( ( row, i ) => {
+		links[ i ] = row.dimensions[ 1 ];
+		return [
+			row.dimensions[ 0 ],
+			Number( row.metrics[ 0 ].values[ 0 ] ).toFixed( 2 ),
+		];
+	} );
+
+	const options = {
+		hideHeader: false,
+		chartsEnabled: false,
+		cap: 5,
+		links,
+	};
+
+	const dataTable = getDataTableFromData( dataMapped, headers, options );
+
+	return (
+		<Layout
```

Comment:
The legacy `Layout` component normally shouldn't be used here, but since this is a data table that has some special formatting ramifications, let's keep it like that for now and cover it via #2022.